cvat/apps/engine/task.py

```diff
@@ -345,6 +347,15 @@ def count_files(file_mapping, counter):
 
     return client_video or server_video, client_archive or server_archive
 
+def _download_data(urls, upload_dir):
+    local_files = []
+    for i, url in enumerate(urls):
+        slogger.glob.info("Downloading: {}".format(url))
+        name = str(i) + '_' + os.path.basename(urlrequest.url2pathname(urlparse.urlparse(url).path))
```

Comment:
It is not good to change the original file name. It will be in a dumped annotation file.

Reply:
Also you should handle download errors better, provide a useful error message.

Reply:
Some URLs don't have an extension. Are you going to support such cases?

For example (first image from google for `cats`), https://img.purch.com/h/1400/aHR0cDovL3d3dy5saXZlc2NpZW5jZS5jb20vaW1hZ2VzL2kvMDAwLzEwNC84MTkvb3JpZ2luYWwvY3V0ZS1raXR0ZW4uanBn

Reply:
> It is not good to change the original file name. It will be in a dumped annotation file.

The index number prefix is to avoid name collision, for example, 2 image files:
http://host/a/1.jpg and http://host/b/1.jpg

> Some URLs don't have an extension. Are you going to support such cases?

Not support. Same as local file which only support known extensions.

Reply:
@zliang7 , I understood that. But there are two ways how to solve the problem correctly (again it is not OK to change original file names because use will use them to match original files):

- Ignore the problem and show a warning
- In the particular case you can use "a/1.jpg" and "b/1.jpg" as names (or even "host/a/1.jpg" and "host/b/1.jpg"). Thus it will be similar to an archive.